crypto/fipsmodule/bn/div_extra.c

```diff
@@ -59,7 +57,64 @@ static uint16_t shift_and_add_mod_u16(uint16_t r, uint32_t a, uint16_t d,
   return t;
 }
 
-uint16_t bn_mod_u16_consttime(const BIGNUM *bn, uint16_t d) {
+
+// Function taken from boringssl's bn.c file.
```

Comment:
Any code that comes from a file X in BoringSSL must be in a file named X in *ring*. Also, the license must be the same. The best way to add functions from bn.c would be to add one commit that is the latest unmodified version from BoringSSL, and then another commit that deletes everything except the license and the code you need.

Note that the licenses are different. That is actually why `div.c` and `div_extra.c` are separate files; the `_extra` files are ISC-licensed, the other files are "OpenSSL licensed," i.e. the bad license everybody is trying to get away from.